salt/modules/zabbix.py

```diff
@@ -1452,3 +1456,109 @@ def hostinterface_update(interfaceid, **connection_args):
             raise KeyError
     except KeyError:
         return ret
+
+
+def template_exists(templateids=None, groupids=None, hostids=None, **connection_args):
+    '''
+    Checks if template that matches the given filter criteria exists.
+
+    :param templateids: template IDs
+    :param groupids: host group IDs
+    :param hostids: host IDs
+    :param _connection_user: Optional - zabbix user (can also be set in opts or pillar, see module's docstring)
+    :param _connection_password: Optional - zabbix password (can also be set in opts or pillar, see module's docstring)
+    :param _connection_url: Optional - url of zabbix frontend (can also be set in opts, pillar, see module's docstring)
+
+    :return: True if at least one host group exists, False if not or on failure.
+
+    CLI Example:
+    .. code-block:: bash
+
+        salt '*' zabbix.template exists templateids=1
+    '''
+    conn_args = _login(**connection_args)
+    try:
+        if conn_args:
+            # support only from Zabbix 2.5 onwards
+            if not templateids:
+                templateids = None
+            if not groupids:
+                groupids = None
+            if not hostids:
+                hostids = None
+            ret = template_get(templateids, groupids, hostids, **connection_args)
+            return bool(ret)
+        else:
+            raise KeyError
+    except KeyError:
+        return False
+
+
+def template_get(templateids=None, groupids=None, hostids=None, **connection_args):
```

Comment:
You want to submit this against the develop branch. This function already exists there, so this will need a conflict merge.